swagger/spec-loader.js

```diff
@@ -117,8 +119,24 @@ function parseSpec(base, spec, log, cb) {
       });
     }
   } else if (spec.swagger === '2.0') {
-    process.nextTick(function() {
-      cb(null, [spec]);
+    // Preserve $ref as $REF after deference
+    spec = lodash.cloneDeepWith(spec, function(o) {
```

Comment:
Hmm..why we don't need the deference for swagger 1.2? 
Just curious since I see the change in test/swagger/pet-store-1.2.json but don't see corresponding code for swagger 1.2.

Reply:
CI also fails on swagger1.2, maybe related:

```
1) loopback:swagger generator creates and configures Pet model from swagger 1.2 spec:

      Uncaught AssertionError: expected [ '#/models/Tag' ] to deeply equal [ 'Tag' ]
      + expected - actual

       [
      -  "#/models/Tag"
      +  "Tag"
       ]
 ```

Reply:
@jannyHou That's due to the dependence on the other PR referenced above.

Reply:
@jannyHou Swagger 1.2 is not supported by `swagger-parser` for dereferencing and it's obsolete.